rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,696 @@
+          description: Internal type definition of the error
+          example: NoSuchNamespaceException
+        code:
+          type: integer
+          minimum: 400
+          maximum: 600
+          description: HTTP response code
+          example: 404
+
+    CatalogConfiguration:
+      type: object
+      description: Server-provided configuration for the catalog.
+      required:
+        - properties
+        - defaults
+      properties:
+        overrrides:
+          type: object
+          nullable: false
+          description:
+            Properties that should be used to override client configuration; applied after defaults and client configuration.
+        defaults:
+          type: object
+          nullable: false
+          description:
+            Properties that should be used as default configuration; applied before client configuration.
+
+    CreateNamespaceRequest:
+      type: object
+      required:
+        - namespace
+      properties:
+        namespace:
+          $ref: '#/components/schemas/Namespace'
+        properties:
+          type: object
+          description: Configured string to string map of properties for the namespace
+          example: '{ "owner": "Hank Bendickson" }'
+          default: '{ }'
+
+    Namespace:
+      description: Reference to one or more levels of a namespace
+      type: array
+      items:
+        type: string
+      example: [ "accounting", "tax" ]
+
+    RenameTableRequest:
+      type: object
+      properties:
+        sourceTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+        destinationTableIdentifier:
+          $ref: '#/components/schemas/TableIdentifier'
+
+    TableIdentifier:
+      type: object
+      required:
+        - namespace
+        - name
+      properties:
+        namespace:
+          type: array
+          description: Individual levels of the namespace
+          items:
+            type: string
+          nullable: false
+        name:
+          type: string
+          nullable: false
+
+    UpdatePropertiesRequest:
+      type: object
+      properties:
+        removables:
+          type: array
+          uniqueItems: true
+          items:
+            type: string
+          example: '[ "department", "access_group" ]'
+        updates:
+          uniqueItems: true
+          type: object
+          items:
+            type: string
+          example: { "owner": "Hank Bendickson" }
+
+
+  #############################
+  # Reusable Response Objects #
+  #############################
+  responses:
+
+    IcebergResponseObject:
+      description: JSON wrapper for all response bodies, with a data object and / or an error object
+      content:
+        application/json:
+          schema:
+            anyOf:
+              - $ref: '#/components/schemas/ResponseDataObject'
+              - $ref: '#/components/schemas/ResponseErrorObject'
+          example: { "data": { ... }, "error": { } }
```

Comment:
This example doesn't appear to be correct. The `ResponseErrorObject` doesn't define an `error` object. It defines `message`, `type`, and `code`.

Maybe it would be good to have this object define the `data` and `error` keys that have types `ResponseDataObject` and `ResponseErrorObject` respectively?

Reply:
Yeah that's a good idea.

Reply:
Done.